### DISCUSSION TITLE
Add support to exclude route from sitemap

### DISCUSSION THREAD

**Discussion by anuj9196** [analogjs/analog#1395] [Ideas]
### Which scope/s are relevant/related to the feature request?

astro-angular

### Information

My SSG application has a few error pages, i.e., the `/404` page.

To generate the SSG of this page, it is required to add it to the route

```typescript
export default defineConfig(({ mode }) => ({
  plugins: [
    analog({
      static: true,
      prerender: {
        routes: async () => [
          '/',
          '/support/',
          '/404'
        ],
        sitemap: {
          host: 'https://example.com/',
        },
      }
    }),
    angular({
      inlineStylesExtension: 'scss',
    }),
  ],
  define: {
    'import.meta.vitest': mode !== 'production',
  },
}));

```

This configuration also adds `/404` to the `sitemap.xml` file which is not required. In the same way, I want to exclude some other pages from the sitemap. However, the `sitemap` configuration only accepts `host`.

A feature can be added to customize or exclude the routes from the `sitemap.xml`.

Also, an option to configure the sitemap file name will be helpful to output the sitemap in another file other than `sitemap.xml`. For example `sitemap-website.xml` or `sitemap-main.xml`, etc.

### Describe any alternatives/workarounds you're currently using

No workaround

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

**Comment by brandonroberts** [comment]
So you want to prerender the `/404` route, but don't want it included in the sitemap? Should it be prerendered then? You can exclude it from the prerender and copy it over manually if that's the case.

Setting a custom filename for the sitemap is reasonable.

**Comment by anuj9196** [comment]
It should be prerendered. Also, I do not want to add many other pages to the `sitemap.xml` file. Like there is `/500` page and a `/404` page on any other pages.

Also `sitemap.xml` file generation is not running `postRenderingHooks`. If a hook can be provided e.g., `postSitemapRenderingHooks`, it can be used to override the content.
